system/security/securesocket/securesocket.cpp

```diff
@@ -801,132 +804,105 @@ int CSecureSocket::wait_read(unsigned timeoutms)
     return m_socket->wait_read(timeoutms);
 }
 
-inline unsigned socketTime(bool useSeconds)
-{
-    if (useSeconds)
-    {
-        time_t timenow;
-        return (unsigned) time(&timenow);
-    }
-    return msTick();
-}
-
-inline unsigned socketTimeRemaining(bool useSeconds, unsigned start, unsigned timeout)
+void CSecureSocket::readtms(void* buf, size32_t min_size, size32_t max_size, size32_t &_sizeRead, unsigned timeoutMs)
```

Comment:
I might be inclined to make this always be nonblocking inside here so we can ensure we never block ?

Reply:
why? If client code using a socket wants it to be blocking, this code should work (and block) but that's okay afaics (depends on the use case).